rs/tests/driver/src/driver/ic_gateway_vm.rs

```diff
@@ -0,0 +1,385 @@
+        let ic_gateway_dir = Path::new(IC_GATEWAY_VMS_DIR).join(name);
+        let ic_gateway_abs_dir = self.get_path(&ic_gateway_dir);
+
+        if !ic_gateway_abs_dir.is_dir() {
+            bail!(
+                "Deployed IC Gateway VM '{}' not found at {}",
+                name,
+                ic_gateway_abs_dir.display()
+            );
+        }
+
+        let playnet_path = ic_gateway_abs_dir.join(PLAYNET_FILE);
+        let playnet: Url = self
+            .read_json_object(&playnet_path)
+            .with_context(|| format!("Failed to read playnet file: {}", playnet_path.display()))?;
+
+        let https_url = playnet
+            .scheme()
+            .eq("https")
+            .then(|| playnet.clone())
+            .context("Expected a TLS URL")?;
+
+        let vm = self
+            .get_deployed_universal_vm(name)
+            .context("Failed to retrieve deployed universal VM")?
+            .get_vm()?;
+
+        Ok(DeployedIcGatewayVm { vm, https_url })
+    }
+
+    fn get_deployed_ic_gateways(&self) -> Result<Vec<DeployedIcGatewayVm>> {
+        let path = self.get_path(IC_GATEWAY_VMS_DIR);
+
+        // no deployed vms
+        if !path.exists() {
+            return Ok(Vec::new());
+        }
+
+        let entries = fs::read_dir(&path)?;
+
+        let mut gateways = Vec::new();
+
+        for entry_result in entries {
+            let entry = entry_result?;
+            if entry.path().is_dir() {
+                let dir_name = entry.file_name().to_string_lossy().to_string();
+                let vm = self.get_deployed_ic_gateway(&dir_name)?;
+                gateways.push(vm);
+            }
+        }
+
+        Ok(gateways)
+    }
+
+    fn write_deployed_ic_gateway(
+        &self,
+        name: &str,
+        playnet: &Url,
+        allocated_vm: &AllocatedVm,
+    ) -> Result<()> {
+        let path = Path::new(IC_GATEWAY_VMS_DIR).join(name);
+        self.write_json_object(path.join(PLAYNET_FILE), playnet)?;
+        self.write_json_object(path.join(IC_GATEWAY_VM_FILE), allocated_vm)
+    }
+}
+
+async fn await_status_is_healthy(logger: &Logger, url: Url, msg: String) -> Result<()> {
+    let client = Client::builder().build()?;
+    let request = Request::new(Method::GET, url);
+    retry_with_msg_async!(&msg, logger, READY_TIMEOUT, RETRY_INTERVAL, || async {
+        let response = client
+            .execute(request.try_clone().unwrap())
+            .await
+            .context("failed to execute request")?;
+        if response.status() == StatusCode::OK {
+            return Ok(());
+        }
+        bail!("ic-gateway not ready yet ...")
+    })
+    .await
+}
+
+pub fn execute_async<F, T>(future: F) -> T
+where
+    F: Future<Output = T>,
+{
+    match Handle::try_current() {
+        Ok(handle) => handle.block_on(future),
+        Err(_) => {
+            let rt = Runtime::new().expect("Failed to create runtime");
+            rt.block_on(future)
+        }
+    }
+}
```

Comment:
Like mentioned above, this can probably be removed in favour of `ic_system_test_driver::utils::block_on`